pkg/virtctl/vnc/vnc.go

```diff
@@ -155,13 +168,37 @@ func (o *VNC) Run(cmd *cobra.Command, args []string) error {
 			glog.Infof("remote-viewer commandline: %v", args)
 		}
 
-		cmnd := exec.Command("remote-viewer", args...)
+		vncBin := ""
+		osType := runtime.GOOS
+		switch osType {
+		case "darwin":
+			if _, err := os.Stat(TIGER_VNC); err == nil {
+				vncBin = TIGER_VNC
+			}
+			if _, err := os.Stat(CHICKEN_VNC); err == nil {
+				vncBin = CHICKEN_VNC
+			}
+			if _, err := os.Stat(REAL_VNC); err == nil {
+				vncBin = REAL_VNC
+				args = append(args, "-WarnUnencrypted=0")
+			}
+		case "linux":
+			vncBin = "remote-viewer"
+		default:
+			// do nothing vncBin = ""
+		}
 
-		output, err := cmnd.CombinedOutput()
-		if err != nil {
-			glog.Errorf("remote-viewer execution failed: %v, output: %v", err, string(output))
+		if vncBin == "" {
+			glog.Errorf("No supported VNC app found in %s", osType)
+			err = fmt.Errorf("No supported VNC app found in %s", osType)
 		} else {
-			glog.V(2).Infof("remote-viewer output: %v", string(output))
+			cmnd := exec.Command(vncBin, args...)
+			output, err := cmnd.CombinedOutput()
+			if err != nil {
+				glog.Errorf("%s execution failed: %v, output: %v", vncBin, err, string(output))
+			} else {
+				glog.V(2).Infof("remote-viewer output: %v", string(output))
+			}
 		}
 		viewResChan <- err
 		close(doneChan)
```

Comment:
the `close(doneChan)` needs to go to the beginning at the function and be `defer close(doneChan)`